pkg/ingester/client/timeseries.go

```diff
@@ -70,27 +70,36 @@ type LabelAdapter labels.Label
 
 // Marshal implements proto.Marshaller.
 func (bs *LabelAdapter) Marshal() ([]byte, error) {
```

Comment:
Please add a test covering `Marshal()` + `Unmarshal()`. Ideally, given a set of input `LabelAdapter`, we should marshal + unmarshal it and ensure the output is identical to the input. This will implicitly test `MarshalToSizedBuffer()` and `Size()`.